apps/internal/base/internal/storage/storage.go

```diff
@@ -84,6 +84,7 @@ func isMatchingScopes(scopesOne []string, scopesTwo string) bool {
 
 // Read reads a storage token from the cache if it exists.
 func (m *Manager) Read(ctx context.Context, authParameters authority.AuthParams, account shared.Account) (TokenResponse, error) {
+	tr := TokenResponse{}
```

Comment:
Do you think there is value in still returning an error in the case where we fail to read some parts of `TokenResponse` to communicate to our caller that we were only able to do a partial read?  In general, I know you shouldn't touch any of the other returned values when error is non nil, but I've seen some go APIs that document what values are correct when an error is returned - I imagine that we could say we could trust an non-zero part of the structure.

Reply:
I'm just worried that some consumer is going to call `Read()` in this state, fail to find an ID token, but since an error is not returned, they go ahead and party on the zero value of the ID Token and expect it to be something sensible.

Reply:
Sure, that's possible in principle, though I think zero values here are obviously invalid. Because this is an internal API, I don't mind trusting callers to continue following the convention of validating cache data before using it (see [`AuthResultFromStorage()`](https://github.com/AzureAD/microsoft-authentication-library-for-go/blob/16c062c187ac55067ece02de9fe65f2f7b4033cd/apps/internal/base/base.go#L99)). If that proves unworkable, I would rather decompose `Read()` into `ReadAccessToken()`, `ReadIDToken()`, etc. than ask callers to ignore irrelevant errors.